### PR TITLE
feat(ffi verifier): added a crude method to pull out CLI arguments, a…

### PR DESCRIPTION
**Pull request by mikegeeves**
…nd make available via FFI

### CODE REVIEW

File: rust/pact_ffi/src/verifier/mod.rs [hunk 1]
```diff
@@ -339,3 +343,176 @@ ffi_fn! {
...
+}
+
+fn parse_argument(long: Option<&str>, short: Option<char>, help: Option<&str>, possible_values: Option<Vec<&str>>, default_value: Option<&OsStr>, multiple: bool, env: Option<(&OsStr, Option<OsString>)>) -> Argument {
+    let mut arg = Argument { short: None, long: None, help: None, possible_values: None, default_value: None, multiple: Some(false), env: None };
+
+    // Long
+    match long {
+        None => {}
+        Some(_val) => {
+            let c_str = CString::new(_val.to_string()).unwrap();
```
Comment (uglyog) [inline rust/pact_ffi/src/verifier/mod.rs]: No need to use CString here. These strings all get put into the final JSON that is sent. All is needed is `arg.long = Some(_val.to_string());`

Variables with underscores are for unused variables where we don't want a compiler warning, so just rename it to `val`.

File: rust/pact_ffi/src/verifier/mod.rs [hunk 2]
```diff
@@ -339,3 +343,176 @@ ffi_fn! {
...
+            arg.help = Some(help.to_string());
+        }
+    }
+
+    // Possible values
+    match possible_values {
+        None => {}
+        Some(_) => {
+            let mut possible_vals: Vec<String> = Vec::new();
+            for possible_val in possible_values.unwrap().iter() {
```
Comment (uglyog) [inline rust/pact_ffi/src/verifier/mod.rs]: You're unwrapping the optional here, when you have already done a pattern match, change it to 
`Some(possible_values) => {` and then remove the unwrap

**Review by uglyog — APPROVED** [review]

File: rust/pact_ffi/src/verifier/mod.rs (hunk 1 above)
Comment (mikegeeves) [reply rust/pact_ffi/src/verifier/mod.rs]: Fixed (that one and the other instances), I think I had gone into Java habits of excessive conversion/casting and failed to tidy after

File: rust/pact_ffi/src/verifier/mod.rs (hunk 2 above)
Comment (mikegeeves) [reply rust/pact_ffi/src/verifier/mod.rs]: Got it, boxing gradually starting to make sense!

**Pull request merged**